### PR TITLE
:seedling: Feature/login

### PR DESCRIPTION
**Pull request by sammysmart95**
- Add Sign In page with full functionality
- Add Sign In hook with full functionality

### CODE REVIEW

**Review by donskilful — CHANGES_REQUESTED** [review]
- print actual error when a user tries to register with existing email
- print an error if the user did not check the check box which says: i agree to terms and conditions 
- Make the password eye checker functional
- After successful sign up redirect to sign in screen

File: src/modules/Auth/hooks/useSignin.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export const useSignIn = () => {
+  const { callApi } = useApi();
+  const { showSuccess, showError } = useAlert();
+  const mutation = useMutation(
+    async (user: SigninInputs) => callApi({ url: '/auth/login', data: user, method: 'POST' }),
+    {
+      onSuccess: async (res) => {
+        // TODO: dispatch all neccesary data
+        console.log('loginRes', res);
+        showSuccess({ title: 'Login Successful!', description: 'You are too good' });
```
[the remark this answers is not in the document]
Comment (sammysmart95) [reply src/modules/Auth/hooks/useSignin.ts:15]: Resolved

File: src/modules/Auth/hooks/useSignin.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      },
+      onError: async (error) => {
+        // TODO: dispatch all neccesary data
+        console.log('loginError', error);
+        showError({ title: 'Login Error', description: "You're a dummy" });
```
[the remark this answers is not in the document]
Comment (sammysmart95) [reply src/modules/Auth/hooks/useSignin.ts:20]: Resolved

**Review by donskilful — CHANGES_REQUESTED** [review]
implement complete design

**Review by donskilful — CHANGES_REQUESTED** [review]
After Successful Sign up it should route to Sign in
If account already exist it does not toast any notification i have to open console to see the error

**Review by donskilful — CHANGES_REQUESTED** [review]
Add an effect on the Sign up and Sing in button when a user clicks and hovers

**Pull request merged**